### PR TITLE
Content: The input of triangular should be at least 2D tensor

### LINKED ISSUE #494: The input of `triangular` should be at least 2D tensor
This is raised by @fdwr on reviewing implementation for `triangular` op on https://github.com/webmachinelearning/webnn-baseline/pull/64. (Thanks Dwayne!)

> it should say the input is at least 2D. You can see other ML libraries just treat each 2D subsection as a separate batch:
> 
> https://numpy.org/doc/stable/reference/generated/numpy.triu.html
> https://pytorch.org/docs/stable/generated/torch.tril.html
> https://onnx.ai/onnx/operators/onnx__Trilu.html
> 
> So shape [4,3,5,7] would apply the triangular masking to the last two dimensions.

### PR DESCRIPTION
**Pull request by inexorabletash**
Fixes #494 which has the text and example. I added the validation step which matches the webnn-baseline impl referenced therein.

* "2-D" seems to be used more commonly than "2D" in the spec
* "batches" doesn't seem to be rigorously defined - room for future improvement here and in matmul?

***
<a href="https://pr-preview.s3.amazonaws.com/inexorabletash/webnn/pull/550.html" title="Last updated on Feb 7, 2024, 5:43 PM UTC (3cef040)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/webnn/550/d1a02f2...inexorabletash:3cef040.html" title="Last updated on Feb 7, 2024, 5:43 PM UTC (3cef040)">Diff</a>

### CODE REVIEW

**Comment by inexorabletash** [comment]
@fdwr and @BruceDai can you do an initial review?

**Comment by fdwr** [comment]
Oh this is great - had it on my queue for a while but haven't popped it 😢. Thanks Joshua. 👏

File: index.bs [hunk 1]
```diff
@@ -6008,7 +6009,16 @@ partial interface MLGraphBuilder {
...
+    const lowerNegative = builder.triangular(input, { upper: false, diagonal: -1 })
+
+    // lower triangular matrix with two batches:
+    //   [[[7, 0, 0],
+    //     [9, 4, 0],
+    //     [2, 6, 3]],
+    //    [[1, 0, 0],
+    //     [4, 5, 0],
+    //     [7, 8, 9]]]
+    const lower = builder.triangular(input, { upper: false });
```
Comment (fdwr) [inline index.bs:6021]: How about `lowerWithBatches` or `batchedLower`?

**Review by fdwr — APPROVED** [review]

File: index.bs (hunk 1 above)
Comment (inexorabletash) [reply index.bs]: `lowerWithBatches` SGTM - done!

Comment (fdwr) [reply index.bs]: > "2-D" seems to be used more commonly than "2D" in the spec

That's amusingly quaint, because nobody really writes "2 dash D" anymore 😉.

"a rank-2 tensor does not usually describe a 2D space."
https://www.tensorflow.org/guide/tensor

"2D convolution layer (e.g. spatial convolution over images)."
https://www.tensorflow.org/api_docs/python/tf/keras/layers/Conv2D

"where is the valid 2D operator"
https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html

"Real-World Examples of 0D, 1D, 2D, 3D, 4D and 5D Tensors"
https://medium.com/data-science-365/real-world-examples-of-0d-1d-2d-3d-4d-and-5d-tensors-100b0837ced4

"This 2D array or tensor ..."
https://wandb.ai/vincenttu/intro-to-tensors/reports/A-Gentle-Intro-To-Tensors-With-Examples--VmlldzozMTQ2MjE5

Comment (inexorabletash) [reply index.bs]: Two dimensions? How quaint!

https://calormen.com/flatland/

Comment (inexorabletash) [reply index.bs]: Spin up a PR to convert the spec and add to conventions?

Comment (fdwr) [reply index.bs]: > Spin up a PR to convert the spec and add to conventions?

Sure, separately from this.

**Review by huningxin — APPROVED** [review]
LGTM, thanks!

**Comment by fdwr** [comment]
> fdwr and BruceDai can you do an initial review?

Bruce is on vacation for Chinese New Year two weeks, but I think he'd be content to merge it as he originally opened the issue on behalf of my comment anyway while reviewing his tests, and Ningxin reviewed it already.

**Comment by anssiko** [comment]
@fdwr please feel free to merge and we’ll check the GH permission knobs are properly set for you at the same time.
